prov/util/src/util_wait.c

```diff
@@ -236,7 +236,7 @@ static int util_wait_fd_try(struct util_wait *wait)
 	}
 	fastlock_release(&wait_fd->lock);
 	ret = fi_poll(&wait->pollset->poll_fid, &context, 1);
-	return (ret > 0) ? -FI_EAGAIN : ret;
+	return (ret > 0) ? -FI_EAGAIN : (ret == -FI_EAGAIN) ? FI_SUCCESS : ret;
```

Comment:
Please update commit message to refer to fi_poll, not poll, since both calls exist